src/util/cilfacade.ml

```diff
@@ -8,11 +8,9 @@ include Cilfacade0
 
 (** Command for assigning an id to a varinfo. All varinfos directly created by Goblint should be modified by this method *)
 let create_var (var: varinfo) =
-  (* TODO Hack: this offset should preempt conflicts with ids generated by CIL *)
-  let start_id = 10_000_000_000 in
+  (* Hack: using negative integers should preempt conflicts with ids generated by CIL *)
   let hash = Hashtbl.hash { var with vid = 0 } in
-  let hash = if hash < start_id then hash + start_id else hash in
-  { var with vid = hash }
+  { var with vid = - hash }
```

Comment:
I'm not sure if there's any guarantee that `Hashtbl.hash` be positive. If it returns something negative, then we'll flip it into a positive `vid` still.

Although I'm now wondering, why we have this business at all. `Cilfacade.create_var` is called with things from `Cil.makeGlobalVar` etc anyway, which already create unique `vid`s that don't conflict with anything else from CIL.

Reply:
The idea is that in a subsequent run of Goblint, the variable will get the same `vid` again, if it is created again.

Reply:
Right, but having our own counter assign `vid`s -1, -2, -3, ... would be fine then? It would avoid the issue of hash being potentially negative.

Reply:
Having our own counter would however require that the variables are created in the same order and this is not necessarily the case, is it?

Reply:
Ah, right, the point is to treat the variable name as unique, such that multiple creations yield the same `vid`. That's also why https://github.com/goblint/analyzer/pull/1312#discussion_r1447416392 had a problem.

Actually [`Hashtbl.hash`](https://v2.ocaml.org/api/Hashtbl.html#VALhash) returns a _non-negative_ integer, so my initial concern is a non-issue. We should have comments about these things though.